android/java/org/chromium/chrome/browser/settings/BraveNewsCategorySources.java

```diff
@@ -0,0 +1,178 @@
+import androidx.preference.SwitchPreferenceCompat;
+import org.chromium.mojo.bindings.ConnectionErrorHandler;
+import org.chromium.mojo.system.MojoException;
+import java.util.Map;
+
+import java.util.TreeMap;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+
+import org.chromium.chrome.R;
+
+public class BraveNewsCategorySources extends PreferenceFragmentCompat implements 
+    ConnectionErrorHandler  {
+
+    private PreferenceManager preferenceManager;
+    private BraveNewsController mBraveNewsController;
+    private String categoryArg;
+
+    private TreeMap <String, List<Publisher>> categsPublishers;
+
+	@Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Bundle args = getArguments();
+        categoryArg = args.getString("category");
+    }
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        InitBraveNewsController();
+        addPreferencesFromResource(R.xml.brave_news_sources_default);
+        getActivity().setTitle(categoryArg);
+
+        categsPublishers = new TreeMap<>();
+        preferenceManager =  getPreferenceManager();
+        mBraveNewsController.getPublishers((publishers) -> {
+            List<Publisher> allPublishers = new ArrayList<>();
+            List<Publisher> categoryPublishers = new ArrayList<>();
+            for (Map.Entry<String,Publisher> entry : publishers.entrySet()) {
+                String key = entry.getKey();
+
+                Publisher publisher = entry.getValue();
+                if (publisher.categoryName.toLowerCase(Locale.ROOT).equals(categoryArg.toLowerCase(Locale.ROOT))){
+                    // categsPublishers.computeIfAbsent(publisher.categoryName, k ->new ArrayList<>()).add(publisher);
+                    categoryPublishers.add(publisher);                    
+                }
+
+                categsPublishers.put(publisher.categoryName, categoryPublishers);                
+                allPublishers.add(publisher);
+
+            }
+            Comparator<Publisher> compareByName = (Publisher o1, Publisher o2) ->
+                    o1.publisherName.compareTo( o2.publisherName );
+            Collections.sort(categoryPublishers, compareByName);
+            Collections.sort(allPublishers, compareByName);
+            categsPublishers.put("All Sources", allPublishers);
+
+            addCategs(categsPublishers);
+        });
+
+    }
+
+    @Override
+    public void onConnectionError(MojoException e) {
+        mBraveNewsController = null;
+        InitBraveNewsController();
+    }
+
+    private void InitBraveNewsController() {
+        if (mBraveNewsController != null) {
+            return;
+        }
+
+        mBraveNewsController =
```

Comment:
`mBraveNewsController.close()` in onDestroy

Reply:
done